pkg/roachprod/roachprod.go

```diff
@@ -733,7 +741,52 @@ func Start(
 	if err != nil {
 		return err
 	}
-	return c.Start(ctx, l, startOpts)
+	if err = c.Start(ctx, l, startOpts); err != nil {
+		return err
+	}
+	updatePrometheusTargets(ctx, l, c)
+	return nil
+}
+
+// UpdateTargets updates prometheus target configurations for a cluster.
+func UpdateTargets(
+	ctx context.Context,
+	l *logger.Logger,
+	clusterName string,
+	clusterSettingsOpts ...install.ClusterSettingOption,
+) error {
+	if err := LoadClusters(); err != nil {
+		return err
+	}
+	c, err := newCluster(l, clusterName, clusterSettingsOpts...)
+	if err != nil {
+		return err
+	}
+	updatePrometheusTargets(ctx, l, c)
+	return nil
+}
+
+// updatePrometheusTargets updates the prometheus instance cluster config. Any error is logged and ignored.
+func updatePrometheusTargets(ctx context.Context, l *logger.Logger, c *install.SyncedCluster) {
+	nodeIPPorts := make([]string, len(c.Nodes))
+	for i, node := range c.VMs {
```

Comment:
Should we add parallelism? I am thinking of a case where `len(c.VMs) >= 50`, and I am not sure what the average latency is for `DiscoverService`. If it's indeed double digit millis (or smaller), then we're fine.

Reply:
I have changed the code to run all in parallel.